test/operators/second_quantization/test_fermionic_op.py

```diff
@@ -141,6 +141,16 @@ def test_init_from_tuple_label(self):
         ]
         self.assertListEqual(actual.to_list(), desired)
 
+    def test_terms(self):
+        """Test terms"""
+        op = FermionicOp([("+", 1.0), ("N", 2.0), ("-+", 3.0)], display_format="dense")
+        expected = [
+            ([("+", 0)], 1.0),
+            ([("+", 0), ("-", 0)], 2.0),
+            ([("-", 0), ("+", 1)], 3.0),
+        ]
+        self.assertEqual(list(op.terms()), expected)
```

Comment:
The expected terms here are in the  same order as the constructor. Is this an expected behavior or might they be in a different order - terms just says it gives an iterable. Would a better test here more say that the set of terms is the same as the set provided such that order is irrelevant? Or is the order part of the expected contract of the terms method?

Reply:
Good catch, the order doesn't matter, so this test should be comparing the sets rather than the lists. However, I can't use `set` here until #607 is merged, so please merge that first.

Reply:
Done.